File: NetFramework/TouchToolkit/VB/TouchToolkitDemo/GcZoomPanelPages/GcZoomPanelDemo.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports C1.Win.TouchToolKit
Imports ClickOnceDemo.C1ZoomPages.Help_Forms

Namespace C1ZoomPanelPages
    Partial Public Class C1ZoomPanelDemo
        Inherits DemoBase
        Public Sub New()
            InitializeComponent()
            InitializeDemo()
        End Sub

        Private Sub InitializeDemo()
            Me.Title = "C1ZoomPanel Basic (Zoom and Scroll)"
            Me.Description = "This page is used to try basic features of C1ZoomPanel. " & vbCr & vbLf & "  - Click the Show Form Button to show a test form. And test the Gestrue zoom/scroll." & vbCr & vbLf & "  - Try to change the settings to knows how to config the C1ZoomPanel."

            AddEnumItemsToComboBox(ZoomPreviewModeComboBox, GetType(ZoomPreviewMode))
            ZoomPreviewModeComboBox.SelectedItem = ZoomPreviewMode.Bitmap
            AddHandler ZoomPreviewModeComboBox.SelectedIndexChanged, AddressOf ZoomPreviewModeComboBox_SelectedIndexChanged

            formatComboBox.Items.Add("Zooming {Percentage}%")
            formatComboBox.Items.Add("Zooming x{ZoomFactor}")
            formatComboBox.SelectedIndex = 0
            AddHandler formatComboBox.TextChanged, AddressOf formatComboBox_TextChanged

            AddHandler FontButton.Click, AddressOf FontButton_Click
            AddHandler ForeColorButton.Click, AddressOf ForeColorButton_Click

            displayLabel.Font = Me.Font
            displayLabel.ForeColor = Me.ForeColor

            boundaryFeedbackUI1.BoundaryFeedbackMode = BoundaryFeedbackMode.Split

            AddEnumItemsToComboBox(scrollIndicatorCombo, GetType(ScrollIndicatorMode))
            scrollIndicatorCombo.SelectedItem = ScrollIndicatorMode.ScrollIndicator

            VerticalRailEnableCheckBox.Checked = True
            HorizontalRailEnableCheckBox.Checked = True
            AllowDoubleTapZoomCheckBox.Checked = False

            AddHandler formButton.Click, AddressOf formButton_Click
        End Sub

        Private Sub formatComboBox_TextChanged(sender As Object, e As EventArgs)
            If formatComboBox.Text IsNot Nothing Then
                displayLabel.Text = formatComboBox.Text.Replace("{Percentage}", "100").Replace("{ZoomFactor}", "1.00")


            End If
        End Sub

        Private Sub formButton_Click(sender As Object, e As EventArgs)
            Using form As New Help()
                FillValues(form.C1ZoomPanel)
                ' The _gcZoom will auto detach when the Form dispose.
                '_gcZoom.SetEnabled(form, false);
                form.ShowDialog(Me)
            End Using
        End Sub

        Private Sub FillValues(gcZoomPanel As C1ZoomPanel)
            gcZoomPanel.ZoomPreviewMode = DirectCast(ZoomPreviewModeComboBox.SelectedItem, ZoomPreviewMode)
            gcZoomPanel.AlternativeContentSettings.Format = formatComboBox.Text
            gcZoomPanel.AlternativeContentSettings.ForeColor = displayLabel.ForeColor
            gcZoomPanel.AlternativeContentSettings.Font = displayLabel.Font

            gcZoomPanel.BoundaryFeedbackMode = boundaryFeedbackUI1.BoundaryFeedbackMode
            gcZoomPanel.ScrollIndicatorMode = DirectCast(scrollIndicatorCombo.SelectedItem, ScrollIndicatorMode)

            gcZoomPanel.IsVerticalRailEnabled = VerticalRailEnableCheckBox.Checked
            gcZoomPanel.IsHorizontalRailEnabled = HorizontalRailEnableCheckBox.Checked
            gcZoomPanel.AllowDoubleTapZoom = AllowDoubleTapZoomCheckBox.Checked
        End Sub

        Private Sub ForeColorButton_Click(sender As Object, e As EventArgs)
            Dim dialog As New ColorDialog()
            If DialogResult.OK = dialog.ShowDialog() Then
                displayLabel.ForeColor = dialog.Color
            End If
        End Sub

        Private Sub FontButton_Click(sender As Object, e As EventArgs)
            Dim dialog As New FontDialog()
            If DialogResult.OK = dialog.ShowDialog() Then

                displayLabel.Font = dialog.Font
            End If
        End Sub

        Private Sub ZoomPreviewModeComboBox_SelectedIndexChanged(sender As Object, e As EventArgs)
            Me.alternativeContentGroup.Visible = False
            Me.NoPreivewInfoPanel.Visible = False
            Me.boundaryFeedbackUI1.Visible = False
            Select Case DirectCast(ZoomPreviewModeComboBox.SelectedItem, ZoomPreviewMode)
                Case ZoomPreviewMode.Bitmap
                    Me.boundaryFeedbackUI1.Visible = True
                    Exit Select
                Case ZoomPreviewMode.NoPreview
                    Me.NoPreivewInfoPanel.Visible = True
                    Exit Select
                Case ZoomPreviewMode.AlternativeContent
                    Me.alternativeContentGroup.Visible = True
                    Exit Select
                Case Else
                    Exit Select
            End Select
        End Sub

        Private Sub AddEnumItemsToComboBox(comboBox As ComboBox, enumType As Type)
            Dim enumArray As Array = [Enum].GetValues(enumType)
            For i As Integer = 0 To enumArray.Length - 1
                comboBox.Items.Add(enumArray.GetValue(i))
            Next
        End Sub
    End Class
End Namespace

File: NetFramework/WinForms/CS/TouchDemo/ZoomPanelPages/GcZoomPanelDemo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using C1.Win.TouchToolKit;
using TouchToolKitDemo.C1ZoomPages.Help_Forms;

namespace TouchToolKitDemo.C1ZoomPanelPages
{
    public partial class C1ZoomPanelDemo : DemoBase
    {
        public C1ZoomPanelDemo()
        {
            InitializeComponent();
            InitializeDemo();
        }

        private void InitializeDemo()
        {
            this.Title = "C1ZoomPanel Basic (Zoom and Scroll)";
            this.Description = @"This page demonstrates how to use C1ZoomPanel’s features and its following properties: ZoomPreviewMode, BoundaryFeedbackMode, AllowDoubleTapZoom, ScrollIndicatorMode, IsHorizontalRailEnabled, IsVerticalRailEnabled, and KeepAspectRation.  Select the settings for these properties on the page and click the Show Form button to see how these settings effect the zooming. When you zoom by touch, the ZoomFactor percentage is shown in the box on the left side of the test form.s";

            AddEnumItemsToComboBox(ZoomPreviewModeComboBox, typeof(ZoomPreviewMode));
            ZoomPreviewModeComboBox.SelectedItem = ZoomPreviewMode.Bitmap;
            ZoomPreviewModeComboBox.SelectedIndexChanged += ZoomPreviewModeComboBox_SelectedIndexChanged;

            formatComboBox.Items.Add("Zooming {Percentage}%");
            formatComboBox.Items.Add("Zooming x{ZoomFactor}");
            formatComboBox.SelectedIndex = 0;
            formatComboBox.TextChanged += formatComboBox_TextChanged;

            FontButton.Click += FontButton_Click;
            ForeColorButton.Click += ForeColorButton_Click;

            displayLabel.Font = this.Font;
            displayLabel.ForeColor = this.ForeColor;

            boundaryFeedbackUI1.BoundaryFeedbackMode = BoundaryFeedbackMode.Split;

            AddEnumItemsToComboBox(scrollIndicatorCombo, typeof(ScrollIndicatorMode));
            scrollIndicatorCombo.SelectedItem = ScrollIndicatorMode.ScrollIndicator;

            VerticalRailEnableCheckBox.Checked = true;
            HorizontalRailEnableCheckBox.Checked = true;
            AllowDoubleTapZoomCheckBox.Checked = false;

            formButton.Click += formButton_Click;
        }

        void formatComboBox_TextChanged(object sender, EventArgs e)
        {
            if (formatComboBox.Text!=null)
            {
                displayLabel.Text = formatComboBox.Text.Replace("{Percentage}", "100").Replace("{ZoomFactor}", "1.00"); ;
            }
        }

        void formButton_Click(object sender, EventArgs e)
        {
            using (Help form = new Help())
            {
                FillValues(form.C1ZoomPanel);
                form.ShowDialog(this);
                // The _C1Zoom will auto detach when the Form dispose.
                //_C1Zoom.SetEnabled(form, false);
            }
        }

        private void FillValues(C1ZoomPanel C1ZoomPanel)
        {
            C1ZoomPanel.ZoomPreviewMode = (ZoomPreviewMode)ZoomPreviewModeComboBox.SelectedItem;
            C1ZoomPanel.AlternativeContentSettings.Format = formatComboBox.Text;
            C1ZoomPanel.AlternativeContentSettings.ForeColor = displayLabel.ForeColor;
            C1ZoomPanel.AlternativeContentSettings.Font = displayLabel.Font;

            C1ZoomPanel.BoundaryFeedbackMode = boundaryFeedbackUI1.BoundaryFeedbackMode;
            C1ZoomPanel.ScrollIndicatorMode = (ScrollIndicatorMode)scrollIndicatorCombo.SelectedItem;

            C1ZoomPanel.IsVerticalRailEnabled = VerticalRailEnableCheckBox.Checked;
            C1ZoomPanel.IsHorizontalRailEnabled = HorizontalRailEnableCheckBox.Checked;
            C1ZoomPanel.AllowDoubleTapZoom = AllowDoubleTapZoomCheckBox.Checked;
        }

        void ForeColorButton_Click(object sender, EventArgs e)
        {
            ColorDialog dialog = new ColorDialog();
            if (DialogResult.OK == dialog.ShowDialog())
            {
                displayLabel.ForeColor = dialog.Color;
            }
        }

        void FontButton_Click(object sender, EventArgs e)
        {
            FontDialog dialog = new FontDialog();
            if (DialogResult.OK == dialog.ShowDialog())
            {
                
                displayLabel.Font = dialog.Font;
            }
        }

        void ZoomPreviewModeComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            this.alternativeContentGroup.Visible = false;
            this.NoPreivewInfoPanel.Visible = false;
            this.boundaryFeedbackUI1.Visible = false;
            switch ((ZoomPreviewMode)ZoomPreviewModeComboBox.SelectedItem)
            {
                case ZoomPreviewMode.Bitmap:
                    this.boundaryFeedbackUI1.Visible = true;
                    break;
                case ZoomPreviewMode.NoPreview:
                    this.NoPreivewInfoPanel.Visible = true;
                    break;
                case ZoomPreviewMode.AlternativeContent:
                    this.alternativeContentGroup.Visible = true;
                    break;
                default:
                    break;
            }
        }

        private void AddEnumItemsToComboBox(ComboBox comboBox, Type enumType)
        {
            Array enumArray = Enum.GetValues(enumType);
            for (int i = 0; i < enumArray.Length; i++)
            {
                comboBox.Items.Add(enumArray.GetValue(i));
            }
        }
    }
}
